tests/h/services/search_index/service_test.py

```diff
@@ -306,3 +298,18 @@ def search_index(es_client, pyramid_request, settings_service, queue):
 @pytest.fixture(autouse=True)
 def es_client():
     return create_autospec(Client, instance=True)
+
+
+@pytest.fixture(autouse=True)
+def indexer(patch):
+    return patch("h.services.search_index.service.indexer")
+
+
+@pytest.fixture(autouse=True)
+def report_exception(patch):
+    return patch("h.services.search_index.service.report_exception")
```

Comment:
`report_exception()` was being used in `service.py` but wasn't patched, which meant that `service_test.py` was executing `h_pyramid_sentry` and `sentry_sdk` code. I don't think `sentry_sdk` tries to access the network as long as the `SENTRY_*` envvars aren't set in the tests. But still we don't want to be calling this code, this import should be patched.

Normally we _wouldn't_ patch more third-party imports but I think we do want to patch `h_pyramid_sentry` / `sentry_sdk`: we already have our own tests for `h_pyramid_sentry` in the `h_pyramid_sentry` project and we don't want h's unit tests to be calling the real `sentry_sdk`

Reply:
You can't move this into `conftest.py` because this patches `"h.services.search_index.service.report_exception"`. When `report_exception()` is imported into some other module you would need a different patch fixture to patch `"h.some.other.module.report_exception"` instead. It'd also differ if something used `import h_pyramid_sentry` as the import style.

If `h_pyramid_sentry` were a `pyramid_services` service then we would be able to have a single centralised mock. The good thing about services is that, by default, none of the services are available to tests. None of the code can access the real services. 'Cos they don't get registered in the tests. Then `conftest.py` can contain centralised mocks for all the services and each test can activate the mocked services that it needs (with a `usefixtures` if it wants). `h_pyramid_sentry` could even provide its own mock service for the apps to use